sources/api/schnauzer/src/helpers.rs

```diff
@@ -1346,9 +1361,188 @@ handlebars_helper!(any_enabled: |arg: Value| {
+/// * `sources/models/src/lib.rs`
+pub fn oci_defaults(
+    helper: &Helper<'_, '_>,
+    _: &Handlebars,
+    _: &Context,
+    renderctx: &mut RenderContext<'_, '_>,
+    out: &mut dyn Output,
+) -> Result<(), RenderError> {
+    // To give context to our errors, get the template name (e.g. what file we are rendering), if available.
+    debug!("Starting oci_defaults helper");
+    let template_name = template_name(renderctx);
+    debug!("Template name: {}", &template_name);
+
+    // Check number of parameters, must be exactly two (OCI spec section to render and settings values for the section)
+    debug!("Number of params: {}", helper.params().len());
+    check_param_count(helper, template_name, 1)?;
+    debug!("params: {:?}", helper.params());
+
+    debug!("Getting the requested OCI spec section to render");
+    let oci_defaults_values = get_param(helper, 0)?;
+    // We want the settings path so we know which OCI spec section we are rendering.
+    // e.g. settings.oci-defaults.resource-limits
+    let settings_path = get_param_key_name(helper, 0)?;
+    // Extract the last part of the settings path, which is the OCI spec section we want to render.
+    let oci_spec_section = settings_path
+        .split('.')
+        .last()
+        .expect("did not find (got None for) an oci_spec_section");
+
+    // Render the requested OCI spec section
+    let section =
+        OciSpecSection::from_str(oci_spec_section).context(error::InvalidOciSpecSectionSnafu)?;
+    let result_lines = match section {
+        OciSpecSection::Capabilities => oci_spec_capabilities(oci_defaults_values)?,
+        OciSpecSection::ResourceLimits => oci_spec_resource_limits(oci_defaults_values)?,
+    };
+
+    // Write out the final values to the configuration file
+    out.write(result_lines.as_str())
+        .context(error::TemplateWriteSnafu {
+            template: template_name.to_owned(),
+        })?;
+
+    Ok(())
+}
+
+/// This helper writes out the capabilities section of the default
+/// OCI runtime spec that `containerd` will use.
+///
+/// This function is called by the `oci_defaults` helper function,
+/// specifically when matching over the `oci_spec_section` parameter
+/// to determine which section of the OCI spec to render.
+///
+/// This helper function generates the linux capabilities section of
+/// the OCI runtime spec from the provided `value` parameter, which is
+/// the settings data from the datastore (`settings.oci-defaults.capabilities`).
+fn oci_spec_capabilities(value: &Value) -> Result<String, RenderError> {
+    let oci_default_capabilities: HashMap<OciDefaultsCapability, bool> =
+        serde_json::from_value(value.clone())?;
+
+    // Only output the capabilities we support
+    let capabilities_lines: Vec<String> = oci_default_capabilities
+        .iter()
+        .filter(|(_, &capability_enabled)| capability_enabled)
+        .map(|(&capability, _)| format!("\"{}\"", capability.to_linux_string()))
+        .collect();
```

Comment:
Is it OK for us to silently disregard unsupported "capabilities"? Whatever they may be?

Reply:
The comment was misleading -- now fixed. What it should have been is "Output the capabilities that are enabled" (coming in the next force push)